src/app/OperationalSessionSetup.cpp

```diff
@@ -292,15 +294,23 @@ void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & ad
     // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
 }
 
-CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessageProtocolConfig & config)
+CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ResolveResult & result)
 {
+    auto & config = result.mrpRemoteConfig;
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
-    // TODO: Combine LargePayload flag with DNS-SD advertisements from peer.
-    // Issue #32348.
     if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload)
     {
-        // Set the transport type for carrying large payloads
-        mDeviceAddress.SetTransportType(chip::Transport::Type::kTcp);
+        if (result.supportsTcpServer)
+        {
+            // Set the transport type for carrying large payloads
+            mDeviceAddress.SetTransportType(chip::Transport::Type::kTcp);
+        }
+        else
+        {
+            // we should not set the large payload while the TCP support is not enabled
+            ChipLogError(Discovery, "LargePayload support is enabled while TCP server support is not enabled");
```

Comment:
This log line needs to include some indication of the peer identity.  Otherwise it's impossible to tell what device it's talking about.

Reply:
done